Nano transactions rewind

gabr1gus opened the pull request:
--criaçao de funçao responsavel por buscar transaçoes que podem ter sido perdidas no momento em que o node esta ligado porem o websocket nao esta ligado
--separado find missing transactions de process transactions e depois descoberto que isso nao é uma ideia muito boa

gabr1gus committed:
movido findMissingTx para novo arquivo

--criado prototipo de findAllMissingTx
--minor changes
--TODO: testar e compilar

gabr1gus committed:
tested & temporary fix

--copiado funçoes necessarias para findAllTx
--atualmente rewindTransactions funciona, porem possui codigo copiado/duplicado de process transaction, o que nao é ideal, em breve isso deverá ser corrigido
--TODO descobrir como compilar esses codigos

GCastilho commented:
Por que não é uma boa ideia?

gabr1gus commented:
> Por que não é uma boa ideia?

porque o rewind é totalmente dependente de parte do codigo do process transaction, copiando as partes necessarias para fazer ele funcionar ficou 80% do process transaction nessa nova funçao, o ideal seria transformar ela ou em parte do process transaction ou criar um novo arquivo para deixar as funçoes comuns entre o process e o rewind

gabr1gus committed:
removido find all missing tx

--pequena reorganizaçao da nova funçao
--precisa de um nome melhor

gabr1gus commented:
aceito sugestoes para o nome da funçao que busca transaçoes perdidas

GCastilho commented:
Pq a rewindTransactions retorna um objeto com a única função dela ao invés de ela _ser_ essa função?

Pq tá assim:

```
function f() {
    const g = () => {}
    return { g }
}
```
ao invés de

```
function f() {
    // código q atualmente está na g
}
```

GCastilho committed:
Reformatado o código horrível do netBoy

-- Reformatado o código, e adicionado coisas simples como espaços entre os = e &&, sabem, essas coisas q todo bom programador deveria fazer
-- Removido alguns ; que não deveriam existir (props pra mim por não ter configurado o eslint no ts)

GCastilho committed:
Refatorado a processTransaction da Nano

-- Refatorado a processTransaction da Nano e retirado a redirectToStd do closure, como ela era chamada apenas uma vez, sua funcionalidade foi implementada diretamente na processTransaction
-- A processTransaction da Nano não é mais uma função geradora que retorna a processTransaction de verdade, já que ela não tem mais nada em seu closure, a função exportada pelo módulo É a processTransaction, o que deixou-a muito mais simples de ser entendida

GCastilho commented:
@gabr1gus tem mais alguma coisa que precisa fazer nesse PR pra implementar o rewind na NANO?

gabr1gus commented:
> @gabr1gus tem mais alguma coisa que precisa fazer nesse PR pra implementar o rewind na NANO?

acho que nao, só nao testei depois das alteraçoes que vc fez, vou testar e fazer esse pull request

gabr1gus commented:
testei, vou marcar como ready for review,

GCastilho reviewed (approved):
Código bonito, código formoso
